crates/astria-composer/src/executor/bundle_factory/mod.rs

```diff
@@ -74,6 +75,23 @@ impl SizedBundle {
         }
     }
 
+    /// Constructs an [`UnsignedTransaction`] from the actions contained in the bundle and `params`.
+    // Method is expected to never panic because only `SequenceActions` are added to the bundle,
+    // which should produce a valid variant of the `ActionGroup` type.
+    #[allow(clippy::panic)]
```

Comment:
Heads-up: with https://github.com/astriaorg/astria/pull/1561 being merged we switch everything to `expect` and require a `reason`:
```suggestion
#[expect(clippy::panic, reason = "method is expected to never panic because only `SequenceActions` are added to the bundle, which should produce a valid variant of the `ActionGroup` type; this is checked by `tests::transaction_construction_should_not_panic")]
```